jetbrains-core/tst/software/aws/toolkits/jetbrains/core/gettingstarted/GettingStartedOnStartupTest.kt

```diff
@@ -0,0 +1,99 @@
+import com.intellij.testFramework.ProjectExtension
+import io.mockk.every
+import io.mockk.junit5.MockKExtension
+import io.mockk.mockkObject
+import io.mockk.verify
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.RegisterExtension
+import software.aws.toolkits.core.utils.deleteIfExists
+import software.aws.toolkits.core.utils.touch
+import software.aws.toolkits.jetbrains.core.credentials.MockCredentialManagerExtension
+import software.aws.toolkits.jetbrains.core.gettingstarted.editor.GettingStartedPanel
+import software.aws.toolkits.jetbrains.settings.GettingStartedSettings
+
+@ExperimentalCoroutinesApi
+@ExtendWith(MockKExtension::class)
+class GettingStartedOnStartupTest {
+    companion object {
+        @JvmField
+        @RegisterExtension
+        val projectExtension = ProjectExtension()
+    }
+
+    @JvmField
+    @RegisterExtension
+    val credManagerExtension = MockCredentialManagerExtension()
+
+    private val sut = GettingStartedOnStartup()
+
+    @Test
+    fun `does not show screen if aws settings exist and has credentials`() {
+        mockkObject(GettingStartedPanel.Companion)
+        every { GettingStartedPanel.openPanel(any()) } returns null
+        GettingStartedSettings.getInstance().shouldDisplayPage = true
+        val fp = getPersistentStateComponentStorageLocation(GettingStartedSettings::class.java) ?: error(
+            "could not determine persistent storage for GettingStartedSettings"
+        )
+        try {
+            fp.touch()
+            sut.runActivity(projectExtension.project)
+        } finally {
+            fp.deleteIfExists()
+        }
+
+        verify(exactly = 0) {
+            GettingStartedPanel.openPanel(projectExtension.project)
+        }
+    }
+
+    @Test
+    fun `does not show screen if has previously shown screen`() {
+        mockkObject(GettingStartedPanel.Companion)
+        every { GettingStartedPanel.openPanel(any()) } returns null
+        GettingStartedSettings.getInstance().shouldDisplayPage = false
+        sut.runActivity(projectExtension.project)
+
+        verify(exactly = 0) {
+            GettingStartedPanel.openPanel(projectExtension.project)
+        }
+    }
+
+    @Test
+    fun `shows screen if aws settings exist and no credentials`() {
+        mockkObject(GettingStartedPanel.Companion)
+        every { GettingStartedPanel.openPanel(any()) } returns null
+        GettingStartedSettings.getInstance().shouldDisplayPage = true
+        val fp = getPersistentStateComponentStorageLocation(GettingStartedSettings::class.java) ?: error(
+            "could not determine persistent storage for GettingStartedSettings"
+        )
+        try {
+            fp.touch()
+            sut.runActivity(projectExtension.project)
+        } finally {
+            fp.deleteIfExists()
+        }
+
+        verify(exactly = 0) {
```

Comment:
Should this be ```
verify {
            GettingStartedPanel.openPanel(projectExtension.project)
        }
```
since we are showing the page?

Reply:
nice catch